### ISSUE TITLE
Spurious warnings for user-defined implicit operators

### ISSUE THREAD

**Issue by bfops** [Microsoft/RoslynClrHeapAllocationAnalyzer#45]
I have this struct:

	public struct ReferenceToAThing
	{
		public string Guid;

		public AThing Instance {
			get;
		}

		public static implicit operator AThing (ReferenceToAThing value);
	}

`AThing` is a reference type, but notice that `ReferenceToAThing` is a value type. The analyzer warns if the implicit operator is used even though it's user-defined and may not actually allocate at all.

**Comment by edespong** [comment]
Hi Zeke, I am trying to reproduce this from you description. 

However, I do not get any warnings. It could be that my understanding of the issue is wrong or that I am building from a newer source than what is currently released. Does the following reproduce the issue for you?

```
namespace TestApplication {
    public class AThing { }

    public struct ReferenceToAThing {
        public AThing Instance { get; }
        public static implicit operator AThing(ReferenceToAThing value) { return value.Instance; }
    }

    class Program1 {
        public static void Main(string[] args) {
            ReferenceToAThing r = new ReferenceToAThing();
            AThing a = r;
        }
    }
}
```

**Comment by bfops** [comment]
I'm seeing the issue:

![error](https://user-images.githubusercontent.com/196249/38153062-a5943a8a-341f-11e8-888e-cc98c05e3811.png)

I'm on version 1.0.0.8 of the plugin. I might be on an artificially old version of the plugin because I'm working in Unity so I'm tethered to an old version of the .NET framework (v3.5, I think).

**Comment by edespong** [comment]
Thanks! Since I do not get the warning for that code I think it will be fixed once a new version gets out. I have not checked, but my guess is that [this fix](https://github.com/Microsoft/RoslynClrHeapAllocationAnalyzer/commit/1bbe15ded5f1e78d392825aee9a38f637f83de92) for issue #19 solved the issue.

**Comment by bfops** [comment]
Sounds good, thanks!
